src/legacy/core_plugins/metrics/public/components/panel_config.js

```diff
@@ -39,9 +41,29 @@ const types = {
 export function PanelConfig(props) {
   const { model } = props;
   const component = types[model.type];
+  const [uiRestrictions, setUIRestrictions] = useState(null);
+
+  useEffect(
+    () => {
+      const visDataSubscription = props.visData$.subscribe(visData =>
+        setUIRestrictions(get(visData, 'uiRestrictions', null))
+      );
+
+      return function cleanup() {
+        visDataSubscription.unsubscribe();
+      };
+    },
+    [props.visData$]
+  );
+
   if (component) {
-    return React.createElement(component, props);
+    return (
+      <UIRestrictionsContext.Provider value={uiRestrictions}>
+        {React.createElement(component, props)}
```

Comment:
I think we could use here the more modern approach of writing it `<Component {...props} />` (I think you will need to name the `component` variable uppercase in this, so that all linters are satisfied.

Reply:
Changed